From 83802669e1eec4bde5bde1f5e87dc7b1ff4e67d9 Mon Sep 17 00:00:00 2001
From: Deepak Sharma <deepak.s27@samsung.com>
Date: Thu, 25 Nov 2021 23:21:40 +0000
Subject: [PATCH 3/3] [Q_asb_2022-01] Fix sorting issue during emergency call
 attempt.

Fix the integer overflow/underflow caused by sorting of duplicate phoneaccounts during emergency call attempt.

Bug: 208267659
Original change: https://android-review.googlesource.com/c/platform/packages/services/Telecomm/+/1903050
Merged-In: I3529e4f86ba5690a8c2e72d6103a84c00bf3f358
Change-Id: I3529e4f86ba5690a8c2e72d6103a84c00bf3f358
Test: Added unit test (see https://googleplex-android-review.git.corp.google.com/c/platform/packages/services/Telecomm/+/16372187)
Test: Manual testing performed to confirm that this corrects repro.
(cherry picked from commit e20b83ec9b3dc2bc5a49a666bd057cf45ce99f15)
(cherry picked from commit 4f64285184fbad85a0a554f3cefa821846be8002)
Merged-In: I3529e4f86ba5690a8c2e72d6103a84c00bf3f358
---
 src/com/android/server/telecom/CreateConnectionProcessor.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/com/android/server/telecom/CreateConnectionProcessor.java b/src/com/android/server/telecom/CreateConnectionProcessor.java
index 6c92724f..89200ad8 100644
--- a/src/com/android/server/telecom/CreateConnectionProcessor.java
+++ b/src/com/android/server/telecom/CreateConnectionProcessor.java
@@ -593,7 +593,7 @@ public class CreateConnectionProcessor implements CreateConnectionResponse {
             }
 
             // then by hashcode
-            return account1.hashCode() - account2.hashCode();
+            return Integer.compare(account1.hashCode(), account2.hashCode());
         });
     }
 
-- 
2.35.1

